src/Traits/DataResolverTraits/DataResolveTrait.php

```diff
@@ -126,27 +126,43 @@ protected function resolveDataObject()
         }
         // Inspect component for element $relation
         if ($this->component->hasMethod($this->columnName)) {
-            $relation = $this->columnName;
-            // We hit a direct method that returns a non-object
-            if (!is_object($this->component->$relation())) {
-                return $this->component->$relation();
-            }
-
-            return self::identify($this->component->$relation(), $this->columns);
+            return $this->getMethodValue();
         }
         // Inspect component has attribute
         if ($this->component->hasField($this->columnName)) {
-            $data = $this->component->{$this->columnName};
-            $dbObject = $this->component->dbObject($this->columnName);
-            if ($dbObject) {
-                // @todo do we need to set the value?
-                $dbObject->setValue($data);
+            return $this->getFieldValue();
+        }
+        $this->cannotIdentifyException($this->component, [$this->columnName]);
+    }
 
-                return self::identify($dbObject, $this->columns);
-            }
+    /**
+     * @return mixed
+     */
+    protected function getMethodValue()
+    {
+        $relation = $this->columnName;
+        // We hit a direct method that returns a non-object
+        if (!is_object($this->component->$relation())) {
+            return $this->component->$relation();
+        }
+
+        return self::identify($this->component->$relation(), $this->columns);
+    }
 
-            return $data;
+    /**
+     * @return mixed
+     */
+    protected function getFieldValue()
+    {
+        $data = $this->component->{$this->columnName};
+        $dbObject = $this->component->dbObject($this->columnName);
+        if ($dbObject) {
+            // @todo do we need to set the value?
```

Comment:
I think the reason why we need to set the value is that we wanted to return in the end with correct datatype.

Reply:
Previous versions of SS4 does now return the correct datatype, like 1 is for Boolean, instead of true.

Reply:
I think this has changed, in recent versions of SS4 but we don't have yet a minimum SS4 version required for this module so this is somewhat like for backwards compatibility.